packages/camera/camera_android_camerax/lib/src/android_camera_camerax.dart

```diff
@@ -137,19 +141,30 @@ class AndroidCameraCameraX extends CameraPlatform {
     startListeningForDeviceOrientationChange(
         cameraIsFrontFacing, cameraDescription.sensorOrientation);
 
-    // Retrieve a ProcessCameraProvider instance.
+    // Retrieve a fresh ProcessCameraProvider instance.
     processCameraProvider ??= await ProcessCameraProvider.getInstance();
+    processCameraProvider!.unbindAll();
 
-    // Configure Preview instance and bind to ProcessCameraProvider.
+    // Configure Preview instance.
+    _resolutionPreset = resolutionPreset;
     final int targetRotation =
         _getTargetRotation(cameraDescription.sensorOrientation);
-    final ResolutionInfo? targetResolution =
+    final ResolutionInfo? previewTargetResolution =
         _getTargetResolutionForPreview(resolutionPreset);
-    preview = createPreview(targetRotation, targetResolution);
-    previewIsBound = false;
-    _previewIsPaused = false;
+    preview = createPreview(targetRotation, previewTargetResolution);
     final int flutterSurfaceTextureId = await preview!.setSurfaceProvider();
 
+    // Configure ImageCapture instance.
+    final ResolutionInfo? imageCaptureTargetResolution =
+        _getTargetResolutionForImageCapture(_resolutionPreset);
+    imageCapture = createImageCapture(null, imageCaptureTargetResolution);
+
+    // Bind configured UseCases to ProcessCameraProvider instance & mark Preview
+    // instance as bound but not paused.
+    camera = await processCameraProvider!
+        .bindToLifecycle(cameraSelector!, <UseCase>[preview!, imageCapture!]);
+    _previewIsPaused = false;
+
```

Comment:
@reidbaker @gmackall @bparrishMines To address @xizhang's [comment](https://github.com/flutter/packages/pull/3287/files#r1123592169), I did some significant refactoring in this method if any of you are willing to take a look.

The main change is that I configured the `Preview` and `ImageCapture` use cases here and bind them to the lifecycle together.